pkg/hints/bigint_hint.go

```
@@ -47,10 +49,142 @@ func NondetBigInt3(virtual_machine VirtualMachine, execScopes ExecutionScopes, i
+	}
+
+	pUncast, err := execScopes.Get("p")
+	if err != nil {
+		return err
+	}
+	p, ok := pUncast.(big.Int)
+	if !ok {
+		return errors.New("Could not cast p value in SafeDivBigint")
+	}
+
+	param_x := new(big.Int).Mul(&res, &y)
+	param_x.Sub(param_x, &x)
+
+	k, err := SafeDivBig(param_x, &p)
+	if err != nil {
+		return err
+	}
+
+	var value big.Int
+	var flag lambdaworks.Felt
+
+	// check if k is positive
+	if k.Cmp(big.NewInt(0)) == 1 {
+		value = *k
+		flag = lambdaworks.FeltFromUint(1)
+	} else {
+		value = *new(big.Int).Neg(k)
+		flag = lambdaworks.FeltFromUint(0)
+	}
+
+	execScopes.AssignOrUpdateVariable("k", *k)
+	execScopes.AssignOrUpdateVariable("value", value)
+
+	val := memory.NewMaybeRelocatableFelt(flag)
+	idsData.Insert("flag", val, vm)
+
+	return nil
+}
+
+func calculateX(vm *VirtualMachine, idsData IdsManager) (big.Int, error) {
+	x_bigint5, err := LimbsFromVarName(5, "x", idsData, vm)
+	if err != nil {
+		return big.Int{}, err
+	}
+	// pack only takes the first three limbs
+	x0 := x_bigint5[0]
+	x1 := x_bigint5[1]
+	x2 := x_bigint5[2]
+
+	var limbs = []lambdaworks.Felt{x0, x1, x2}
+
+	x_lower := BigInt3{Limbs: limbs}
+	x_lower_bigint := x_lower.Pack86()
+
+	d3 := x_bigint5[3].ToSigned()
+	d4 := x_bigint5[4].ToSigned()
+
+	base := BASE()
+	d3.Mul(d3, base.Exp(&base, big.NewInt(3), nil))
+	d4.Mul(d4, base.Exp(&base, big.NewInt(4), nil))
+
+	x_lower_bigint.Add(&x_lower_bigint, d3)
+	x_lower_bigint.Add(&x_lower_bigint, d4)
+
+	return x_lower_bigint, nil
+}
+
+func bigintPackDivMod(vm *VirtualMachine, execScopes *ExecutionScopes, idsData IdsManager) error {
+
+	pUnpack, err := BigInt3FromVarName("P", idsData, vm)
+	if err != nil {
+		return err
+	}
+	p := pUnpack.Pack86()
+
+	x, err := calculateX(vm, idsData)
+	if err != nil {
+		return err
+	}
+	x1 := x
+
+	yUnpacked, err := BigInt3FromVarName("y", idsData, vm)
+	if err != nil {
+		return err
+	}
+	y := yUnpacked.Pack86()
+
+	res, _ := math_utils.DivMod(&x1, &y, &p)
```

Dont ignore the error